pkg/cmd/server/config.go

```diff
@@ -211,6 +216,9 @@ func (cfg Config) GetKubernetesPublicAddress() (*url.URL, error) {
 }
 
 func (cfg Config) GetAssetPublicAddress() (*url.URL, error) {
+	if cfg.AssetPublicAddr.Provided {
+		return cfg.AssetPublicAddr.URL, nil
+	}
 	// Derive the asset public address by incrementing the master public address port by 1
 	t, err := cfg.GetMasterPublicAddress()
```

Comment:
Seems like the default port for the assetPublicAddr should be assetBindAddr port.  That would allow it to do match defaulting for masterPublicAddr (directly specified, masterAddr if specified, bindAddr).